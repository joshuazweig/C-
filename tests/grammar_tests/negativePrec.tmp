ACCEPT
[program:
  [decls:
    [decls:]
    [fdecl:
      [typ: INT]
      ID
      LPAREN
      [formals_opt:]
      RPAREN
      LBRACE
      [combined_list:
        [combined_list:
          [combined_list:
            [combined_list: [combined_list:] [vdecl: [typ: INT] ID SEMI]]
            [vdecl: [typ: INT] ID SEMI]
          ]
          [stmt:
            [expr:
              ID
              ASSIGN
              [expr: [expr: MINUS [expr: LITERAL]] STAR [expr: LITERAL]]
            ]
            SEMI
          ]
        ]
        [stmt: [expr: ID ASSIGN [expr: MINUS [expr: ID]]] SEMI]
      ]
      RBRACE
    ]
  ]
  EOF
]
