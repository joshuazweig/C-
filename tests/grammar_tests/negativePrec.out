ACCEPT
[program:
  [decls:
    [decls:]
    [fdecl:
      [typ: INT]
      ID
      LPAREN
      [formals_opt:]
      RPAREN
      LBRACE
      [vdecl_list:
        [vdecl_list: [vdecl_list:] [vdecl: [typ: INT] ID SEMI]]
        [vdecl: [typ: INT] ID SEMI]
      ]
      [stmt_list:
        [stmt_list:
          [stmt_list:]
          [stmt:
            [expr:
              ID
              ASSIGN
              [expr: [expr: MINUS [expr: LITERAL]] STAR [expr: LITERAL]]
            ]
            SEMI
          ]
        ]
        [stmt: [expr: ID ASSIGN [expr: MINUS [expr: ID]]] SEMI]
      ]
      RBRACE
    ]
  ]
  EOF
]
